src/lib/response-triggers/userlocation.ts

```diff
@@ -0,0 +1,334 @@
+		}
+
+		await this.dbLocation.setUserLocation(userId, location);
+	}
+
+	private async deleteUserLocationAdmin(args : Argument[], message : MessageObject) : Promise<string> {
+		if (args.length < 4) {
+			return 'Please specify a list of mentions/me';
+		}
+
+		const users = args.filter((item, index) => index > 2 );
+
+		if (users.length === 1 && !users[0].isMention) {
+			if (users[0].text !== 'me') {
+				return `Invalid person ${users[0].text}. Either specify 'me' or mention a person`;
+			}
+
+			try {
+				await this.dbLocation.deleteUserLocation(message.personId);
+				return 'Location has been deleted';
+			} catch (error) {
+				Logger.error(`[UserLocation.setUserLocation] Error deleting location for user ${message.personId}: ${error.message}`);
+				return 'Error deleting location';
+			}
+		}
+
+		if (users.some(x => !x.isMention)) {
+			return 'Mixed mentions and non mentions not allowed';
+		}
+
+		const usersPromise = users.map(async x => {
+			try {
+				await this.dbLocation.deleteUserLocation(x.userId);
+				return {
+					user: x.text,
+					success: true
+				};
+			} catch (error) {
+				Logger.error(`[UserLocation.setUserLocation] Error deleting location for user ${x.userId}: ${error.message}`);
+				return {
+					user: x.text,
+					success: false
+				}
+			}
+		});
+
+		const result = await Promise.all(usersPromise);
+		if (result.some(x => !x.success)) {
+			return `Location deleting was unsuccessful for user(s): ${result.filter(x => !x.success).map(x => `'${x.user}'`).join(', ')}`;
+		}
+
+		return 'Location has been deleted';
+	}
+
+	private async getAllUserLocationMessage() : Promise<string> {
+		let userLocations : UserLocationRow[];
+		try {
+			userLocations = await this.dbLocation.getAllUserLocations();
+		} catch(e) {
+			Logger.error(`[UserLocation.getUserLocationMessage] Error getting user locations: ${e.message}`);
+			return 'Error getting user locations';
+		}
+
+		if (userLocations.length === 0) {
+			return 'No user locations set';
+		}
+
+		const mappedPromise = userLocations.map(async x => {
+			const user = await this.dbUsers.getUser(x.userid);
+			return {
+				username: user.username,
+				location: x.location
+			};
+		});
+
+		const mapped = await Promise.all(mappedPromise);
+
+		const columnWidths = this.getColumnWidths(mapped);
+
+		let message = 'Here is the current config:\n```\n';
+
+		message += tableHelper.padString('Username', columnWidths.username) + ' | Location\n';
+		message += ''.padEnd(columnWidths.username, '-') + '-+-' + ''.padEnd(columnWidths.location, '-') + '\n';
+
+		for (const config of userLocations) {
+			const user = await this.dbUsers.getUser(config.userid);
+			message += user.username.padEnd(columnWidths.username) + ' | ' + config.location + '\n';
+		}
+
+		message += '```';
+
+		return message;
+	}
+
+	private getColumnWidths(values : { username: string; location: string; }[]) : { username : number, location : number } {
```

Comment:
This stuff normally lives in tableHelper.ts, and this file is long enough as it is

Reply:
I mean does it really belong in tablehelper.ts? I get the sentiment but also all of those function really only get used in one spot. I can move it though

Reply:
There does feel like there is some common code we could refactor out - eg getting the column widths for each individual property on its own

Reply:
Did a reafactor, see what you think